cmake/PythonDynamicHelper.cmake

```diff
@@ -0,0 +1,126 @@
+# =============================================================================
+# Configure support for dynamic Python to use multiple Python versions
+# =============================================================================
+# ~~~
+# NEURON can be built with python modules that can be usable from multiple
+# versions of Python. Here we check if NRN_ENABLE_PYTHON_DYNAMIC is valid
+# and determine an include directory for version 2 and/or 3 to build
+# libnrnpython<major>.so. Depending on the pythons used, either or both of
+# NRNPYTHON_INCLUDE3 or NRNPYTHON_INCLUDE2 will be defined.
+#
+# The above is good for mac and linux. Sadly, for MINGW, a distinct
+# NRNPYTHON_INCLUDE<major><minor> is needed for each python in the
+# NRN_PYTHON_DYNAMIC list. This is because libnrnpython<major><minor>.dll
+# must be linked against the specfic libpython to avoid undefined name errors.
+# Thus, at least for MINGW, parallel to the NRN_PYTHON_DYNAMIC list
+# we construct the lists NRN_PYTHON_VER_LIST, NRN_PYTHON_INCLUDE_LIST,
+# and NRN_PYTHON_LIB_LIST
+# ~~~
+
+set(LINK_AGAINST_PYTHON ${MINGW})
```

Comment:
Although only MINGW required that libnrniv.so link against python,
I like that we have an independent name for this that exactly describes what is being done.
It allows one to also try it out on linux/mac (by modifyin that .cmake file). And it remains unclear whether it will be needed
for certain  strategies for creating pip wheels.

Reply:
Thanks for the explanation, I agree with that.